File: src/main/java/phoenix/client/render/TankRenderer.java
package phoenix.client.render;

import com.mojang.blaze3d.matrix.MatrixStack;
import net.minecraft.client.renderer.IRenderTypeBuffer;
import net.minecraft.client.renderer.tileentity.TileEntityRenderer;
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
import phoenix.client.models.block.TankModel;
import phoenix.tile.redo.TankTile;

public class TankRenderer extends TileEntityRenderer<TankTile>
{
    public TankRenderer(TileEntityRendererDispatcher rendererDispatcherIn)
    {
        super(rendererDispatcherIn);
    }

    @Override
    public void render(TankTile te, float partialTicks, MatrixStack matrixStackIn, IRenderTypeBuffer bufferIn, int combinedLightIn, int combinedOverlayIn)
    {
        new TankModel(te).render(matrixStackIn, bufferIn, combinedLightIn, combinedOverlayIn);
    }
}

File: src/main/kotlin/phoenix/client/render/TankRenderer.kt
package phoenix.client.render

import com.mojang.blaze3d.matrix.MatrixStack
import net.minecraft.client.renderer.IRenderTypeBuffer
import net.minecraft.client.renderer.tileentity.TileEntityRenderer
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher
import phoenix.client.models.block.TankModel
import phoenix.tile.redo.TankTile

class TankRenderer<T : TankTile>(rendererDispatcherIn: TileEntityRendererDispatcher) : TileEntityRenderer<T>(rendererDispatcherIn)
{
    override fun render(te: T, partialTicks: Float, matrixStackIn: MatrixStack, bufferIn: IRenderTypeBuffer, combinedLightIn: Int, combinedOverlayIn: Int) = TankModel(te).render(matrixStackIn, bufferIn, combinedLightIn, combinedOverlayIn)
}